presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/EncoderErrorCode.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder;
+
+import io.prestosql.spi.ErrorCode;
+import io.prestosql.spi.ErrorCodeSupplier;
+import io.prestosql.spi.ErrorType;
+
+import static io.prestosql.spi.ErrorType.EXTERNAL;
+
+public enum EncoderErrorCode
+        implements ErrorCodeSupplier
+{
+    /** A requested data conversion is not supported */
+    ENCODER_CONVERSION_NOT_SUPPORTED(0, EXTERNAL);
+
+    private final ErrorCode errorCode;
+
+    EncoderErrorCode(int code, ErrorType type)
+    {
+        errorCode = new ErrorCode(code + 0x0101_0000, name(), type);
```

Comment:
make sure you are using unique code:

```
kokosing@m16:~/presto$ git grep '0x0101_0000'
presto-record-decoder/src/main/java/io/prestosql/decoder/DecoderErrorCode.java:        errorCode = new ErrorCode(code + 0x0101_0000, name(), type);
```

Why do you use `io.prestosql.plugin.kafka.KafkaErrorCode`?

Reply:
I deleted `EncoderErrorCode` because I don't use the error code in this pr, I think it might be better to just add a `KAFKA_ENCODER_ERROR` to `KafkaErrorCode` in the future